diff --git a/arch/arm/dts/sun50i-h616.dtsi b/arch/arm/dts/sun50i-h616.dtsi
index f9ae55a472..5a2007488e 100644
--- a/arch/arm/dts/sun50i-h616.dtsi
+++ b/arch/arm/dts/sun50i-h616.dtsi
@@ -1063,6 +1063,13 @@
 			#phy-cells = <0>;
 		};
 
+    hdmi_audio: hdmi-audio@1 {
+			compatible = "allwinner,sunxi-hdmi-machine";
+			sunxi,cpudai-controller = <&ahub_i2s1>;
+			sunxi,snddaudio-codec = "hdmi-audio-codec.4.auto";
+			status = "okay";
+		};
+
 		tcon_top: tcon-top@6510000 {
 			compatible = "allwinner,sun50i-h6-tcon-top";
 			reg = <0x06510000 0x1000>;
