diff --git a/arch/arm/dts/sun50i-h616-orangepi-zero2.dts b/arch/arm/dts/sun50i-h616-orangepi-zero2.dts
index cb8600d0ea..2d7c6ba877 100644
--- a/arch/arm/dts/sun50i-h616-orangepi-zero2.dts
+++ b/arch/arm/dts/sun50i-h616-orangepi-zero2.dts
@@ -86,7 +86,7 @@
 };
 
 &mmc0 {
-	vmmc-supply = <&reg_dcdce>;
+	vmmc-supply = <&reg_aldo1>;
 	cd-gpios = <&pio 5 6 GPIO_ACTIVE_LOW>;	/* PF6 */
 	bus-width = <4>;
 	status = "okay";
@@ -118,6 +118,7 @@
 
 		regulators {
 			reg_aldo1: aldo1 {
+				compatible = "regulator-fixed";
 				regulator-always-on;
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
@@ -207,7 +208,7 @@
 &pio {
 	vcc-pc-supply = <&reg_aldo1>;
 	vcc-pf-supply = <&reg_aldo1>;
-	vcc-pg-supply = <&reg_bldo1>;
+	vcc-pg-supply = <&reg_aldo1>;
 	vcc-ph-supply = <&reg_aldo1>;
 	vcc-pi-supply = <&reg_aldo1>;
 };
